### PR TITLE
Fix doctest for `TFDeiTForImageClassification`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Since TF 2.10, `tf.random.set_seed` with a fixed seed won't give the same model weights anymore. See the [release note](https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0). We need `tf.keras.utils.set_random_seed()` for this purpose.

This PR fixes the doctest for `TFDeiTForImageClassification` by using the above solution.

- I have to update the expected value however.
- I get the new expected value on a CPU VM. It should work on the GPU VM too, but let's keep an eye on the CI result.

### CODE REVIEW

**Review by Rocketknight1 — APPROVED** [review]
LGTM!

**Review by gante — APPROVED** [review]
LGTM! 👍 

is the line that sets the non-Keras seed still needed? (`>>> tf.random.set_seed(3)`)

**Comment by ydshieh** [comment]
> LGTM! 👍
> 
> is the line that sets the non-Keras seed still needed? (`>>> tf.random.set_seed(3)`)

No we don't need it :-) Removed it.
